#! /bin/sh /usr/share/dpatch/dpatch-run
## ppdc-dynamic-linking.dpatch by Sune Vuorela <debian@pusling.com>
##
## DP: Dynamically link ppdc, to work around segfault on mipsen.
## DP: Debian #548246

@DPATCH@
diff -urNad cups-1.4.1~/ppdc/Makefile cups-1.4.1/ppdc/Makefile
--- cups-1.4.1~/ppdc/Makefile	2009-05-27 01:27:06.000000000 +0200
+++ cups-1.4.1/ppdc/Makefile	2009-10-13 09:29:29.859258406 +0200
@@ -237,14 +237,14 @@
 # genstrings - generate GNU gettext strings.
 #
 
-genstrings:		genstrings.o libcupsppdc.a ../cups/libcups.a \
+genstrings:		genstrings.o libcupsppdc.so ../cups/libcups.a \
 			sample.drv ../data/media.defs
 	echo Linking $@...
 	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o genstrings genstrings.o \
-		libcupsppdc.a ../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) \
+		-lcupsppdc ../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) \
 		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
 	echo Generating localization strings...
-	./genstrings >sample.c
+	LD_LIBRARY_PATH=.:../cups/ ./genstrings >sample.c
 
 
 #
@@ -258,12 +258,12 @@
 
 ppdc-static:		ppdc.o libcupsppdc.a  ../cups/libcups.a foo.drv foo-fr.po
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdc-static ppdc.o libcupsppdc.a \
+	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdc-static ppdc.o -lcupsppdc \
 		../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	echo Testing PPD compiler...
-	./ppdc-static -l en,fr -I ../data foo.drv
-	./ppdc-static -l en,fr -z -I ../data foo.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en,fr -I ../data foo.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -l en,fr -z -I ../data foo.drv
 
 
 #
@@ -286,14 +286,14 @@
 
 ppdi-static:		ppdc-static ppdi.o libcupsppdc.a  ../cups/libcups.a
 	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdi-static ppdi.o libcupsppdc.a \
+	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o ppdi-static ppdi.o -lcupsppdc \
 		../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	echo Testing PPD importer...
 	$(RM) -r ppd ppd2 sample-import.drv
-	./ppdc-static -I ../data sample.drv
-	./ppdi-static -I ../data -o sample-import.drv ppd/*
-	./ppdc-static -I ../data -d ppd2 sample-import.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -I ../data sample.drv
+	LD_LIBRARY_PATH=.:../cups/ ./ppdi-static -I ../data -o sample-import.drv ppd/*
+	LD_LIBRARY_PATH=.:../cups/ ./ppdc-static -I ../data -d ppd2 sample-import.drv
 	if diff -qr ppd ppd2; then \
 		echo PPD import OK; \
 	else \
@@ -326,7 +326,7 @@
 
 testcatalog:		testcatalog.o libcupsppdc.a ../cups/libcups.a
 	echo Linking $@...
-	$(CXX) $(LDFLAGS) -o $@ testcatalog.o libcupsppdc.a \
+	$(CXX) $(LDFLAGS) -o $@ testcatalog.o -lcupsppdc \
 		../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 
